bin/ch/ch.cpp

```diff
@@ -196,48 +207,536 @@ static void CALLBACK PromiseContinuationCallback(JsValueRef task, void *callback
+
+    wcscat_s(*res, rlength, uri + fpos);
+}
+
+void GetDefaultTTDDirectory(char16** res, const char16* optExtraDir)
+{
+#ifndef _WIN32
+    *res = nullptr;
+    AssertMsg(false, "Not XPLAT yet.");
+#else
+    char16* path = new char16[MAX_PATH];
+    path[0] = _u('\0');
+
+    GetModuleFileName(NULL, path, MAX_PATH);
+
+    char16* spos = wcsstr(path, _u("\\Build\\VcBuild\\"));
+    AssertMsg(spos != nullptr, "Something got renamed or moved!!!");
+
+    int ccount = (int)((((byte*)spos) - ((byte*)path)) / sizeof(char16));
+
+    *res = (char16*)CoTaskMemAlloc(MAX_PATH * sizeof(char16));
+    if(*res == nullptr)
+    {
+        //This is for testing only so just assert and return here is ok
+        AssertMsg(false, "OOM");
+        return;
+    }
+
+    (*res)[0] = _u('\0');
+
+    for(int i = 0; i < ccount; ++i)
+    {
+        (*res)[i] = path[i];
+    }
+    (*res)[ccount] = _u('\0');
+
+    wcscat_s(*res, MAX_PATH, _u("\\test\\_ttdlog\\"));
+
+    if(wcslen(optExtraDir) == 0)
+    {
+        wcscat_s(*res, MAX_PATH, _u("_defaultLog"));
     }
     else
     {
-        runScript = ChakraRTInterface::JsRunScript(fileContents, WScriptJsrt::GetNextSourceContext(), fullPathWide, nullptr /*result*/);
+        wcscat_s(*res, MAX_PATH, optExtraDir);
     }
 
-    free(fullPathWide);
+    bool isPathDirName = ((*res)[wcslen(*res) - 1] == _u('\\'));
+    if(!isPathDirName)
+    {
+        wcscat_s(*res, MAX_PATH, _u("\\"));
+    }
+
+    delete[] path;
+#endif
+}
 
-    if (runScript != JsNoError)
+static void CALLBACK GetTTDDirectory(const char16* uri, char16** fullTTDUri)
+{
+#ifndef _WIN32
+    *fullTTDUri = nullptr;
+    AssertMsg(false, "Not XPLAT yet.");
+#else
+    if(uri[0] != _u('!'))
     {
-        WScriptJsrt::PrintException(fileName, runScript);
+        bool isPathDirName = (uri[wcslen(uri) - 1] == _u('\\'));
+
+        size_t rlength = (wcslen(uri) + 2);
+        *fullTTDUri = (wchar_t*)CoTaskMemAlloc(rlength * sizeof(char16));
+        if(*fullTTDUri == nullptr)
+        {
+            //This is for testing only so just assert and return here is ok
+            AssertMsg(false, "OOM");
+            return;
+        }
+
+        (*fullTTDUri)[0] = _u('\0');
+
+        wcscat_s(*fullTTDUri, rlength, uri);
+        if(!isPathDirName)
+        {
+            wcscat_s(*fullTTDUri, rlength, _u("\\"));
+        }
     }
     else
     {
-        // Repeatedly flush the message queue until it's empty. It is necessary to loop on this
-        // because setTimeout can add scripts to execute.
-        do
+        GetDefaultTTDDirectory(fullTTDUri, uri + 1);
+    }
+#endif
+}
+
+static void CALLBACK TTInitializeForWriteLogStreamCallback(const char16* uri)
+{
+    //If the directory does not exist then we want to write it
+    CreateDirectoryIfNeeded(uri);
+
+    //Clear the logging directory so it is ready for us to write into
+    DeleteDirectory(uri);
+}
+
+static HANDLE TTOpenStream_Helper(const char16* uri, bool read, bool write)
```

Comment:
For the current time this is true but we may want to allow rw opens later as well. I added an Assert to ensure current use is correct and make this explicit in the code.